cmdx.py

```diff
@@ -4004,6 +4290,19 @@ def _encode1(path):
 
     mobject = om1.MObject()
     selectionList.getDependNode(0, mobject)
+
+    if "." in path:  # costs 0.2µs
+        # check whether path is a plug or a component
+        try:
+            mdag = om1.MDagPath()
+            cp = om1.MObject()
+            selectionList.getDagPath(0, mdag, cp)
+            return (mdag, cp)
+        except:
+            mplug = om1.MPlug()
+            mplug = selectionList.getPlug(0, mplug)
+            return mplug
+
```

Comment:
Did this need to change? It's only used internally, is anything using the added features?

Reply:
Oh no it's a scar from a previous implementation test of component with legacy support of ObjectSet

I'll rollback this. Especially if we try to avoid legacy in the future..